Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentEnd/FocusHelper.js

```diff
@@ -2,24 +2,37 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-(function() {
- "use strict";
- const createFormElementEventHandler = (eventType) => {
-  return (event) => {
-    const elementType = event.target.nodeName;
-    if (elementType === "INPUT" || elementType === "TEXTAREA" || event.target.isContentEditable) {
-      webkit.messageHandlers.focusHelper.postMessage({ eventType, elementType });
+"use strict";
+
+const isButton = (element) => {
+  if (element.nodeName !== "INPUT") {
+    return false;
+  }
+
+  return (element.type == "BUTTON" || element.type == "SUBMIT");
+};
+
+const handler = (event) => {
+  const eventType = event.type;
+  const elementType = event.target.nodeName;
+  // We can receive focus and blur events from `a` elements and anything with a `tabindex` attribute.
+  // We should also not fire for buttons..
+  if (elementType === "INPUT" || elementType === "TEXTAREA" || event.target.isContentEditable) {
+    if (!isButton(event.target)) {
+      webkit.messageHandlers.focusHelper.postMessage({ 
+        eventType, 
+        elementType 
+      });
     }
-  };
- };
+  }
+};
 
- const options = {
+const options = {
   capture: true,
   passive: true,
- };
+};
 
- const body = window.document.body;
- ["focus", "blur"].forEach((eventType) => {
-  body.addEventListener(eventType, createFormElementEventHandler(eventType), options);
- });
-})();
+const body = window.document.body;
+["focus", "blur"].forEach((eventType) => {
+  body.addEventListener(eventType, handler, options);
```

Comment:
nit: its probably *slightly* more inclusive (safer) to just attach this directly to `document`. Browser rendering engines are very forgiving of out-of-spec HTML and it might be possible for a `<textarea>` outside of the `<body>` tag to get rendered.